web/siteplan/src/util/Controls/MeasureControl.ts

```diff
@@ -0,0 +1,219 @@
+      stroke: new Stroke({
+        color: 'rgba(0, 0, 0, 0.7)'
+      }),
+      fill: new Fill({
+        color: 'rgba(255, 255, 255, 0.2)'
+      })
+    })
+  })
+
+  constructor (map: Map) {
+    const button = document.createElement('button')
+    button.innerHTML = '&#128207;' // HTML Encoding for "📏"
+    button.className = 'measure-control-button'
+    button.title = 'Messwerkzeug'
+
+    const element = document.createElement('div')
+    element.className = 'measure-control-container ol-unselectable ol-control'
+    element.appendChild(button)
+
+    super({ element })
+    this.vectorLayer = this.createVectorLayer()
+    this.view = map.getView()
+    this.map = map
+    this.view.set(MeasureControl.MEASURE_CONTROL_KEY, this)
+    this.drawLineInteraction = this.getDrawMeasureLineInteraction()
+
+    button.addEventListener('click', () => {
+      if (store.state.measureEnable ) {
+        this.onDeactiveMeasure()
+      } else {
+        this.onActiveMeasure()
+      }
+
+      store.commit('setMeasureEnable', !store.state.measureEnable)
+    })
+  }
+
+  getDrawMeasureLineInteraction () {
+    let vectorSource = this.vectorLayer.getSource()
+    if (!vectorSource) {
+      vectorSource = new VectorSource()
+      this.vectorLayer.setSource(vectorSource)
+    }
+
+    const drawInteraction = new Draw({
+      source: vectorSource,
+      type: 'LineString',
+      style: feature => {
+        console.log('TEST')
+        const geometryType = feature.getGeometry()?.getType()
+        if (geometryType == 'LineString' || geometryType === 'Point') {
+          return this.lineStyle
+        }
+      }
+    })
+
+    drawInteraction.on('drawstart', event => this.onStartDraw(event))
+    drawInteraction.on('drawend', () => this.onDrawEnd())
+    return drawInteraction
+  }
+
+  onStartDraw (event: DrawEvent) {
+    this.sketch = event.feature
+    const geometry = this.sketch.getGeometry()
+    if (!geometry) {
+      return
+    }
+
+    this.listerner = geometry.on('change', e => {
+      const geom = e.target
+      let output
+      let tooltipCoord
+      if (geom instanceof LineString) {
+        output = this.formatLength(geom)
+        tooltipCoord = geom.getLastCoordinate()
+      }
+
+      if (!this.measureTooltipElement) {
+        return
+      }
+
+      this.measureTooltipElement.innerHTML = output ?? 'length'
+      this.measureTooltip.setPosition(tooltipCoord)
+    })
+  }
+
+  onDrawEnd () {
+    if (!this.listerner) {
+      return
+    }
+
+    if (this.measureTooltipElement != null) {
+      this.measureTooltipElement.className = 'ol-tooltip ol-tooltip-static'
+    }
+
+    this.measureTooltip?.setOffset([0, -7])
+    this.sketch = null
+    this.measureTooltipElement = null
+    this.craeteMeasureTooltip()
+    unByKey(this.listerner)
+  }
+
+  formatLength (line: LineString): string {
+    const length = getLength(line)
+    return Math.round(length * 100) / 100 + ' ' + 'm'
+  }
+
+  craeteMeasureTooltip () {
```

Comment:
@TruongQuangSB small typo 

```suggestion
  createMeasureTooltip () {
```